nesc/whip6/platforms/parts/mcu/cc26xx/timers/HalEventCountPrv.nc

```diff
@@ -110,9 +111,19 @@ implementation {
         uint64_t ov_value;
         uint32_t cnt_value;
         atomic {
-            cnt_value = getTimerValue(which);
-            if (cnt_value < lastRead[getCntIdx(which)])
-                overflows[getCntIdx(which)] += MAX_COUNT;
+            cnt_value = getTimerValue(which) & (MAX_COUNT - 1);
+            if (which & TIMER_A) {
+                if (call ChannelAInterrupt.getPending() |
+                        (TimerIntStatus(call CC26xxTimer.base(), TRUE) & TIMER_CAPA_MATCH)) {
+                    cnt_value = getTimerValue(TIMER_A) | MAX_COUNT;
+                }
+            }
+            if (which & TIMER_B) {
+                if (call ChannelBInterrupt.getPending() |
+                        (TimerIntStatus(call CC26xxTimer.base(), TRUE) & TIMER_CAPB_MATCH)) {
+                    cnt_value = getTimerValue(TIMER_B) | MAX_COUNT;
+                }
+            }
             lastRead[getCntIdx(which)] = cnt_value;
```

Comment:
Why do we need the lastRead table?

Reply:
to know the value if the timer is stopped

Reply:
OK. Add a comment to the sources.

Reply:
done